core/server/common/src/main/java/alluxio/Registry.java

```diff
@@ -91,81 +93,82 @@ public MasterRegistry() {}
   }
 
   /**
-   * @param clazz the class of the master to add
-   * @param master the master to add
-   * @param <T> the type of the master to add
+   * @param clazz the class of the server to add
+   * @param server the server to add
+   * @param <U> the type of the server to add
    */
-  public <T> void add(Class<T> clazz, Master master) {
+  public <U> void add(Class<U> clazz, T server) {
     try (LockResource r = new LockResource(mLock)) {
-      mRegistry.put(clazz, master);
+      mRegistry.put(clazz, server);
       mCondition.signalAll();
     }
   }
 
   /**
-   * @return a list of all the registered masters, order by dependency relation
+   * @return a list of all the registered servers, order by dependency relation
    */
-  public List<Master> getMasters() {
-    List<Master> masters = new ArrayList<>(mRegistry.values());
-    Collections.sort(masters, new DependencyComparator());
-    return masters;
+  public List<T> getServers() {
+    List<T> servers = new ArrayList<>(mRegistry.values());
+    Collections.sort(servers, new DependencyComparator());
+    return servers;
   }
 
   /**
-   * Starts all masters in dependency order. If A depends on B, A is started before B.
+   * Starts all servers in dependency order. If A depends on B, A is started before B.
    *
-   * If a master fails to start, already-started masters will be stopped.
+   * If a server fails to start, already-started servers will be stopped.
    *
-   * @param isLeader whether to start the masters as leaders
+   * @param options the start options
    * @throws IOException if an IO error occurs
    */
-  public void start(boolean isLeader) throws IOException {
-    List<Master> started = new ArrayList<>();
-    for (Master master : getMasters()) {
+  public void start(Object options) throws IOException {
+    List<T> servers = new ArrayList<>();
+    for (T server : getServers()) {
       try {
-        master.start(isLeader);
-        started.add(master);
+        server.start(options);
+        servers.add(server);
       } catch (IOException e) {
-        for (Master startedMaster: started) {
-          startedMaster.stop();
-          throw e;
+        for (T started : servers) {
+          started.stop(options);
         }
+        throw e;
       }
     }
   }
 
   /**
-   * Stops all masters in reverse dependency order. If A depends on B, B is stopped before A.
+   * Stops all servers in reverse dependency order. If A depends on B, B is stopped before A.
    *
+   * @param options the stop options
    * @throws IOException if an IO error occurs
    */
-  public void stop() throws IOException {
-    for (Master master : Lists.reverse(getMasters())) {
-      master.stop();
+  public void stop(Object options) throws IOException {
```

Comment:
I would strongly prefer not to include `Object options` in this signature. It's untyped, it adds complexity, and it's always null. Are there advantages to this besides mirroring the signature of `start`?